javatests/arcs/android/host/AndroidAllocatorTest.kt

```diff
@@ -100,68 +100,56 @@ open class AndroidAllocatorTest : AllocatorTestBase() {
         super.setUp()
     }
 
-    @Ignore("b/154947390 - Deflake")
     @Test
     fun allocator_startArc_throwsException() = runAllocatorTest {
         writingHost().throws = true
 
         assertSuspendingThrows(ArcHostException::class) {
-            allocator.startArcForPlan(
-                "readWriteParticle",
-                PersonPlan
-            )
+            allocator.startArcForPlan(PersonPlan).waitForStart()
         }
     }
 
-    @Ignore("b/154947390 - Deflake")
+    @Ignore("b/155502365 - Broken by inability to dereference nested arc->particle->handle entity")
     @Test
     override fun allocator_doesntCreateArcsOnDuplicateStartArc() {
         super.allocator_doesntCreateArcsOnDuplicateStartArc()
     }
 
-    @Ignore("b/154947390 - Deflake")
     @Test
     override fun allocator_startFromOneAllocatorAndStopInAnother() {
         super.allocator_startFromOneAllocatorAndStopInAnother()
     }
 
-    @Ignore("b/154947390 - Deflake")
     @Test
     override fun allocator_verifyStorageKeysNotOverwritten() {
         super.allocator_verifyStorageKeysNotOverwritten()
     }
 
-    @Ignore("b/154947390 - Deflake")
     @Test
     override fun allocator_verifyArcHostStartCalled() {
         super.allocator_verifyArcHostStartCalled()
     }
 
-    @Ignore("b/154947390 - Deflake")
     @Test
     override fun allocator_restartArcInTwoExternalHosts() {
         super.allocator_restartArcInTwoExternalHosts()
     }
 
-    @Ignore("b/154947390 - Deflake")
     @Test
     override fun allocator_canStartArcInTwoExternalHosts() {
         super.allocator_canStartArcInTwoExternalHosts()
     }
 
-    @Ignore("b/154947390 - Deflake")
     @Test
     override fun allocator_computePartitions() {
         super.allocator_computePartitions()
     }
 
-    @Ignore("b/154947390 - Deflake")
     @Test
     override fun allocator_verifyStorageKeysCreated() {
         super.allocator_verifyStorageKeysCreated()
     }
 
-    @Ignore("b/154947390 - Deflake")
```

Comment:
Were you able to verify that these are now fixed by running the tests something like 100+ times?

Reply:
Unfortunately, not totally fixed. :(  When I ran them 20 times, they were fixed, but with 100, I got 1 out of 100 flaky, and 3 out of 100 flaky on two runs. Most of the errors are always

"java.lang.IllegalStateException: processModelUpdate on closed StorageProxy
	at arcs.core.storage.StorageProxy.processModelUpdate(StorageProxy.kt:298)
	at arcs.core.storage.StorageProxy.access$processModelUpdate(StorageProxy.kt:39)
	at arcs.core.storage.StorageProxy$onMessage$2$5.invoke(StorageProxy.kt:268)
	at arcs.core.storage.StorageProxy$onMessage$2$5.invoke(StorageProxy.kt:39)
	at arcs.core.util.Scheduler$Task.invoke(Scheduler.kt:153)
	at arcs.core.util.Scheduler.executeAgenda(Scheduler.kt:139)
	at arcs.core.util.Scheduler$startProcessingJob$2$1.invokeSuspend(Scheduler.kt:98)
	at arcs.core.util.Scheduler$startProcessingJob$2$1.invoke(Scheduler.kt)
	at "

So there is still so weird condition where the handles are closed and then a pending operation from the scheduler runs. Can we just make EntityManager.close() cancel() the Scheduler or pause() it?

Reply:
Now down to 3 failures out of 1200. Landing it like this, but can disable again if it starts failing.